macros/Read_building_permits.sas

```diff
@@ -0,0 +1,135 @@
+/**************************************************************************
+ Program:  Read_building_permits.sas
+ Library:  DCRA
+ Project:  NeighborhoodInfo DC
+ Author:   Rob Pitingolo
+ Created:  06/20/17
+ Version:  SAS 9.4
+ Environment:  Windows
+ 
+ Description:  Macro to read Building permit data for a given year.
+
+ Modifications:
+**************************************************************************/
+
+%macro read_building_permits (year=, revisions=);
+
+/* Read in permits shapefile */
+proc mapimport out=permits_&year._in
+	datafile="&_dcdata_r_path\DCRA\Raw\Building_Permits_in_&year..shp";
+ 	select applicatio desc_of_wo fees_paid fee_type full_addre issue_date lastmodifi maraddress owner_name
+		   permit_app permit_cat permit_id permit_sub permit_typ ssl xcoord ycoord zoning objectid;
+	rename applicatio=AppStatus desc_of_wo=DescOfWork fees_paid=Fees fee_type=FeeType full_addre=Address issue_date=IssueDate
+		   lastmodifi=LastModified maraddress=MARAddress owner_name=OwnerName permit_app=ApplicantName permit_cat=PermitCategory
+		   permit_id=PermitID permit_sub=PermitSubcategory permit_typ=PermitType;
+run;
+
+proc freq data = permits_&year._in;
+	tables PermitType;
+run;
+
+/* Read in Census block shapefile */
+proc mapimport out=block_shp_in
+  datafile="&_dcdata_r_path\OCTO\Maps\Census_Blocks__2010.shp";  
+run;
+
+proc sort data=block_shp_in out=block_shp (keep = x y geoid10); 
+	by geoid10;
+run;
+
+
+/* Spatial join permits to Census blocks */
+proc ginside includeborder
+  data=permits_&year._in 
+  map=block_shp
+  out=permits_&year._join;
+  id geoid10;
+run;
+
+
+/* Add standard geographies to each permit, create dummy vars and label */
+data permits_&year._geo;
+	set permits_&year._join;
+
+	geoblk2010 = geoid10;
+
+	%Block10_to_anc02;
+	%Block10_to_anc12;
+	%Block10_to_bpk;
+	%Block10_to_city;
+	%Block10_to_cluster00;
+	%Block10_to_cluster_tr00;
+	%Block10_to_cluster17;
+	%Block10_to_eor;
+	%Block10_to_psa04;
+	%Block10_to_psa12;
+	%Block10_to_stantoncommons;
+	%Block10_to_tr00;
+	%Block10_to_tr10;
+	%Block10_to_vp12;
+	%Block10_to_ward02;
+	%Block10_to_ward12;
+	%Block10_to_zip;
+
+	label x = "Latitude (decimal degrees)"
+		  y = "Longitude (decimal degrees)"
+		  geoblk2010 = "Census block (2010)"
+		  appstatus = "Permit application status"
+		  descofwork = "Description of work"
+		  fees = "Amount of permit fees ($)"
+		  feetype = "Description and amount of fees required for permit"
+		  address = "Address on permit application"
+		  issuedate = "Permit issue date"
```

Comment:
*issuedate* and *lastmodified* should be reformatted as SAS date or datetime values.